src/Search/Search.tsx

```diff
@@ -1,5 +1,37 @@
-import React from "react";
-import * as ReactDOM from "react-dom";
+import React, { useState, useEffect } from "react";
 import './Search.scss'
 
+
+const Search: React.FC = () => {
+  const [query, setQuery] = useState<string>('')
+  const [debouncedQuery, setDebouncedQuery] = useState<string>(query)
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
```

Comment:
This is the timeout function for the search component that won't send a request until user is done typing for 750 ms

Reply:
I like this a lot, removes a lot of the chaos of instantaneous searching